src/posix/platform/include/openthread/platform/secure_settings.h

```diff
@@ -0,0 +1,178 @@
+/*
+ *  Copyright (c) 2021, The OpenThread Authors.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions are met:
+ *  1. Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *  2. Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *  3. Neither the name of the copyright holder nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ * @brief
+ *   This file includes platform abstraction for secure non-volatile storage of settings.
+ */
+
+#ifndef OPENTHREAD_POSIX_SECURE_SETTINGS_H_
+#define OPENTHREAD_POSIX_SECURE_SETTINGS_H_
+
+#include <openthread/instance.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @addtogroup plat-settings
+ *
+ * @brief
+ *   This module includes the platform abstraction for secure non-volatile storage of settings.
+ *
+ * @{
+ *
+ */
+
+/**
+ * This function performs any initialization for the secure settings subsystem, if necessary.
+ *
+ * @param[in]  aInstance The OpenThread instance structure.
+ *
+ */
+void otPlatSecureSettingsInit(otInstance *aInstance);
```

Comment:
If I understand the code, these functions are intended to be used by the platform layer only, and not directly called by OpenThread core itself (i.e. OT code will only call/use `otPlatSettingsSetCriticalKeys()` to indicate which keys are critical and should use be saved differently). 
- Is this correct? 
- If so, I think it's better not to use `otPlat` prefix in the name of the functions here.  
- Generally, `otPlat{}` is used for platform layer functions/definitions that are intended for direct use by OT core code.

Reply:
Yes, the APIs `otPlatSecureSettingsXxx` will only be used by the posix platform layer. I will change them from 'otPlatSecureSettingsXxx' to `otPosixSecureSettingsXxx`.

Reply:
BTW, I think the method name should be `platformSecureSettingXxx`

Reply:
The APIs `otPosixSecureSettingsXxx` will be implemented by the Vendor. They may locate in another repo. So I think it is better to use the name `otPosixSecureSettingsXxx`. Comments?